canonical-web-and-design/maas-ui
# fix(ui): fetch VMs when viewing resources card

## Done

- Fetch VMs when viewing resources card or NUMA card. I'm surprised this bug has lasted this long tbh

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to the resources tab of a KVM and refresh the page
- Check that a loading spinner shows while the machines load, then eventually the dropdown shows.

## Fixes

Fixes #3105
